bigquery/google/cloud/bigquery/dbapi/exceptions.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2016 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""Exceptions used in the Google BigQuery DB-API."""
```

Comment:
This module is so WAT to me. @jonparrott @tseaver @lukesneeringer is it common for a PEP to be so opinionated about errors like this?

----

Directly from the PEP:

> This is the exception inheritance layout:
> 
> ```
> StandardError
> |__Warning
> |__Error
>   |__InterfaceError
>    |__DatabaseError
>       |__DataError
>       |__OperationalError
>       |__IntegrityError
>       |__InternalError
>       |__ProgrammingError
>       |__NotSupportedError
> ```